lib/rgb_matrix/animation/solid_reactive.ex

```diff
@@ -69,31 +68,40 @@ defmodule RGBMatrix.Animation.SolidReactive do
     %{tick: tick, color: color, leds: leds, hits: hits} = state
     %{speed: _speed, distance: distance} = config
 
-    {colors, hits} =
-      Enum.map_reduce(leds, hits, fn led, hits ->
-        case hits do
-          %{^led => {hit_tick, direction_modifier}} ->
-            # TODO: take speed into account
-            if tick - hit_tick >= distance do
-              {{led.id, color}, Map.delete(hits, led)}
-            else
-              hue = mod(color.h + (tick - hit_tick - distance) * direction_modifier, 360)
-              {{led.id, HSV.new(hue, color.s, color.v)}, hits}
-            end
-
-          _else ->
-            {{led.id, color}, hits}
-        end
+    colors =
+      Enum.map(leds, fn
+        led when is_map_key(hits, led) ->
+          {hit_tick, direction_modifier} = hits[led]
+
+          if tick - hit_tick >= distance do
+            {led.id, color}
+          else
+            hue_shift = (tick - hit_tick - distance) * direction_modifier
+            hue = mod(color.h + hue_shift, 360)
+            {led.id, HSV.new(hue, color.s, color.v)}
+          end
+
+        led ->
+          {led.id, color}
       end)
 
-    # FIXME: leaves color 1 away from base
-    # TODO: we can optimize this by rewriting the above instead of filtering here:
-    colors =
-      Enum.filter(colors, fn {_id, this_color} ->
-        this_color != color
+    updated_hits =
+      Enum.reduce(hits, hits, fn {led, {hit_tick, _direction_modifier}}, hits_acc ->
+        if tick - hit_tick >= distance do
+          Map.delete(hits_acc, led)
+        else
+          hits_acc
+        end
       end)
```

Comment:
:information_source: (optional)

You could also use [`Enum.reject`](https://hexdocs.pm/elixir/Enum.html#reject/2) here to avoid:
1. Using `hits` as both the enumerable and the accumulator
2. The `if` statement
3. The need for `Map.delete`

```suggestion
    updated_hits =
      hits
      |> Enum.reject(fn {_led, {hit_tick, _direction_modifier}} ->
        tick - hit_tick >= distance
      end)
      |> Enum.into(%{})
```

Reply:
I did actually mean to have hits be both enumerable and accumulator, but this is much cleaner!

I haven't used `Enum.reject` before, that's really nice.